src/connectors/voice-search/__tests__/connectVoiceSearch-test.js

```diff
@@ -65,6 +68,28 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/voice-searc
         })
       );
     });
+
+    it('creates custom voice helper', () => {
+      const voiceHelper = {
+        isBrowserSupported: () => true,
+        dispose: () => {},
+        getState: () => ({
+          isSpeechFinal: true,
+          status: 'askingPermission',
+          transcript: '',
+        }),
+        isListening: () => true,
```

Comment:
`isListening` seems to share lots of similarities with state variables, like `isSpeechFinal`. Wdyt about having `isListening` in `getState`?

Reply:
Our implementation for `isListening()` is 

```js
  const isListening = (): boolean =>
    state.status === 'askingPermission' ||
    state.status === 'waiting' ||
    state.status === 'recognizing';
```

For our case, we could have it as a boolean variable in `getState`.
However, in terms of using third-party helpers, `isListening` might have to be a function on their side (idk).
And `isListening, startListening, stopListening` sounds consistent to me.

Reply:
In the custom helper I wrote the condition is

```js
  const isListening = () =>
    state.status === 'askingPermission' || state.status === 'recognizing';
```

since there's no 'waiting' phase. Other custom flows will have their own statuses as well

Reply:
Right! I don't have a complete understanding of ways to use the voice search widget so I'm only suggesting ideas, they might not make total sense—feel free to ignore.